src/java/org/apache/cassandra/utils/memory/MemtablePool.java

```diff
@@ -47,23 +48,25 @@
 
     final WaitQueue hasRoom = new WaitQueue();
 
-    MemtablePool(long maxOnHeapMemory, long maxOffHeapMemory, float cleanThreshold, Runnable cleaner)
+    MemtablePool(long maxOnHeapMemory, long maxOffHeapMemory, float cleanThreshold, MemtableCleaner cleaner)
     {
+        Preconditions.checkArgument(cleaner != null, "Cleaner should not be null");
+
         this.onHeap = getSubPool(maxOnHeapMemory, cleanThreshold);
         this.offHeap = getSubPool(maxOffHeapMemory, cleanThreshold);
         this.cleaner = getCleaner(cleaner);
```

Comment:
If `cleaner` can't be `null`, should we just inline the `MemtableCleanerThread` creation here? Also, there are some places below where we still check to see if `cleaner` is `null` that are unnecessary now?

Reply:
While I am with you for simplifying and optimizing the code, I would prefer not to refactor this as part of this patch.